Sources/PubNubChat/PubNub/Models/ChatMessage.swift

```diff
@@ -56,8 +56,8 @@ public struct ChatMessage<Custom: ChatCustomData>: Identifiable, Hashable, Codab
       id: String = UUID().uuidString,
```

Comment:
Maybe creating protocol for MessagePayload and setting this structure behind could remove generics and let developer to create his own custom structure with forced by protocol fields and custom one added by him.

Reply:
I think for now we want to be consistent with how we handle dynamic objects within our entities, but it would be nice to investigate the pros/cons of that type of interface.